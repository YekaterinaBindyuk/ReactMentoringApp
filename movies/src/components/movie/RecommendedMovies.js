import React from "react";
import MoviesList from "../movies-list/MoviesList";
import { RECOMMENDED_MOVIES } from "../../environment/const";
import actionCreators from "../../redux/action-creators/actionCreators";
import { connect } from "react-redux";

export class RecommendedMovies extends React.Component {
  componentDidUpdate(prevProps) {
    const { getRecommendedMovies, genres } = this.props;
    if (genres && this.props.genres !== prevProps.genres) {
      getRecommendedMovies(genres);
    }
  }

  render = () => {
    const { moviesList } = this.props;

    if (moviesList.length) {
      return (
        <div className="recommended">
          {RECOMMENDED_MOVIES}
          <MoviesList moviesList={moviesList} />
        </div>
      );
    } else return null;
  };
}

const mapDispatchToProps = {
  getRecommendedMovies: actionCreators.getRecommendedMovies
};

const mapStateToProps = state => {
  return {
    moviesList: state.recommendedMovies
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(RecommendedMovies);
