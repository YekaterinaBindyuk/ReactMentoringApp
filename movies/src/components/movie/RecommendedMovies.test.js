import React from "react";
import { render, mount } from "enzyme";
import { RecommendedMovies } from "./RecommendedMovies";

describe("RecommendedMovies", () => {
  it('should render correctly in "debug" mode', () => {
    const moviesList = [
      {
        budget: 0,
        genres: ["Comedy"],
        id: 447365,
        overview: "The third film based on Marvel's Guardians of the Galaxy.",
        poster_path:
          "https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg",
        release_date: "2020-05-01",
        revenue: 0,
        runtime: null,
        tagline: "",
        title: "Guardians of the Galaxy Vol. 3",
        vote_average: 0,
        vote_count: 9
      }
    ];
    const component = render(<RecommendedMovies moviesList={moviesList} />);
    expect(component).toMatchSnapshot();
  });
});

describe("RecommendedMovies", () => {
  it("should get recommended movies on update", () => {
    const mockGetRecommendedMoviesfn = jest.fn();

    const props = {
      moviesList: [],
      getRecommendedMovies: mockGetRecommendedMoviesfn,
      genres: ["Comedy"]
    };
    const component = mount(
      <RecommendedMovies  {...props}
      />
    );
    component.setProps({ genres: ["Comedy", "Action"] });
    expect(mockGetRecommendedMoviesfn).toHaveBeenCalled();
  });
});
