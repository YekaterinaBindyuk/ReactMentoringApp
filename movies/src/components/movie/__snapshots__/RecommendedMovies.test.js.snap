// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecommendedMovies should render correctly in "debug" mode 1`] = `
<div
  class="recommended"
>
  Recommended movies
  <div
    class="movies-list"
  >
    <div
      class="movies-list-item"
    >
      <img
        alt="Guardians of the Galaxy Vol. 3"
        class="img_small"
        src="https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg"
      />
      <h3
        class="title"
      >
        Guardians of the Galaxy Vol. 3
      </h3>
      <section>
        <div
          class="genres"
        >
          Comedy
        </div>
      </section>
      <div>
        2020-05-01
      </div>
    </div>
  </div>
</div>
`;
